src/components/language-switch/language-switch.tsx

```diff
@@ -1,44 +1,115 @@
 import { useI18next } from 'gatsby-plugin-react-i18next';
 import React from 'react';
-import styled from 'styled-components';
-import { Link } from '../links/links';
+import styled, { DefaultTheme } from 'styled-components';
+import { Chevron } from '../icons/chevron';
 
-const StyledLanguageLink = styled(Link)<{ selected?: boolean }>`
-  margin: 1px 0 1px 12px;
+interface LanguageSwitchProps {
+  translation: any;
+  className?: string;
+  $lightTheme?: boolean;
+  fromNavigation?: boolean;
+}
+
+const getTextColor = ({
+  fromNavigation,
+  $lightTheme,
+  theme,
+}: {
+  fromNavigation?: boolean;
+  $lightTheme?: boolean;
+  theme: DefaultTheme;
+}) => {
+  if (fromNavigation || $lightTheme) {
+    return theme.palette.text.default;
+  } else {
+    return theme.palette.text.header;
+  }
+};
+
+const getTextHoverColor = ({
+  fromNavigation,
+  $lightTheme,
+  theme,
+}: {
+  fromNavigation?: boolean;
+  $lightTheme?: boolean;
+  theme: DefaultTheme;
+}) => {
+  if (fromNavigation) {
+    return theme.palette.text.header;
+  } else if (!$lightTheme) {
+    return theme.palette.text.headerHover;
+  }
+};
+
+const StyledNav = styled.nav<{
+  $lightTheme?: boolean;
+  fromNavigation?: boolean;
+}>`
+  transition: color 0.2s;
+  color: ${(props) => getTextColor(props)};
+  &:hover {
+    color: ${(props) => getTextHoverColor(props)};
+  }
+`;
+
+const StyledSelection = styled.select<{
+  $lightTheme?: boolean;
+  fromNavigation?: boolean;
+}>`
```

Comment:
This styled component does not make use of any of the props so this generic type is dead code. I suggest to remove it:
```suggestion
const StyledSelection = styled.select`
```